src/txdb.cpp

```diff
@@ -716,21 +734,32 @@ bool CBlockTreeDB::LoadBlockIndexGuts()
                 pindexNew->nTime          = diskindex.nTime;
                 pindexNew->nBits          = diskindex.nBits;
                 pindexNew->nNonce         = diskindex.nNonce;
-                pindexNew->nSolution      = diskindex.nSolution;
+                // the Equihash solution will be loaded lazily from the dbindex entry
                 pindexNew->nStatus        = diskindex.nStatus;
                 pindexNew->nCachedBranchId = diskindex.nCachedBranchId;
                 pindexNew->nTx            = diskindex.nTx;
                 pindexNew->nSproutValue   = diskindex.nSproutValue;
                 pindexNew->nSaplingValue  = diskindex.nSaplingValue;
                 pindexNew->segid          = diskindex.segid;
                 pindexNew->nNotaryPay     = diskindex.nNotaryPay;
-                // Consistency checks
-                auto header = pindexNew->GetBlockHeader();
-                if (header.GetHash() != pindexNew->GetBlockHash())
-                    return error("LoadBlockIndex(): block header inconsistency detected: on-disk = %s, in-memory = %s",
-                                 diskindex.ToString(),  pindexNew->ToString());
+
                 if ( 0 ) // POW will be checked before any block is connected
```

Comment:
So by disabling this code we avoid reading nSolution which we would not need here.
As this check `header.GetHash() != pindexNew->GetBlockHash()` seems not make much sense as the header is constructed from the block index.
And, CheckProofOfWork will be done anyway when the best chain is activated,
Is this right?

Reply:
Detailed explanation of consistency checks removal exists here: https://github.com/zcash/zcash/issues/6532. The [commit](https://github.com/Alrighttt/komodo/pull/42/commits/688de6060cd9f94269604d7ab9078d8a262ee977) just moved these checks to `if (0) { ... }` block instead of removal and do nothing with `CheckProofOfWork`, as it already was there (in unreachable block of code).

Reply:
Okay seems my guess is similar to that explanation.
(Yes I see that zcash calls CheckProofOfWork() there for the disk error reasons but we did not for long time already. I hope we do not lose anything due to that)